vm/devices/user_driver/src/memory.rs

```diff
@@ -157,4 +157,39 @@ impl MemoryBlock {
     pub fn offset_in_page(&self) -> u32 {
         self.base as u32 % PAGE_SIZE as u32
     }
+
+    /// Returns base address of the memory block.
+    pub fn base_va(&self) -> u64 {
+        self.base as u64
+    }
+}
+
+pub mod save_restore {
```

Comment:
This module should be in `fixed_pool_alloc`.

Reply:
I did that at first, but then can not resolve circular dependencies. I had to find a place for it where it can be accessed by LoadedVm module and also thinking if this can become more generic save/restore for other memory types.

Reply:
I don't see the circular dependency. This is only used by `underhill_core` and `fixed_pool_alloc`.

Reply:
Missing `pub use` was the key to solve this.